test/cmdlineTests/ast_ir_undeployable_contract/args

```diff
@@ -0,0 +1 @@
+--ir-ast-json --ir-optimized-ast-json --optimize --pretty-json --json-indent 4
```

Comment:
Maybe it would be good to also add a test for `--ir` and `--ir-optimized` (or maybe just all outputs) with undeployable contract. Your PR changes its behavior, but I do not see any existing tests affected by this so it's apparently not covered in any way.